### ISSUE TITLE
For loops with parentheses

### ISSUE THREAD

**Issue by phillipdonald** [phalcon/php-zephir-parser#3]
### Expected and Actual Behavior

> **Expecting syntax of for loops with parentheses to work as they do with if statements and while loops.**

```php
class ForParentheses
{
    public function alpha(string $str)
    {
        char $ch;

        for ($ch in $str) {
            echo $ch, "\n";
        }
    }
}
```

> **Build output.**
```php
vagrant@zephir:~/zephir/utils$ zephir build 
Zephir\Exception: Cannot parse file: /home/vagrant/zephir/utils/utils/forparentheses.zep
```
### Details
```
ubuntu-16.04

vagrant@zephir:~/zephir/utils$ php -v
PHP 7.0.15-0ubuntu0.16.04.4 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.15-0ubuntu0.16.04.4, Copyright (c) 1999-2017, by Zend Technologies
vagrant@zephir:~/zephir/utils$

vagrant@zephir:~/zephir/utils$ zephir
Zephir version 0.9.7-6608033934

Compiled from source.

```

**Comment by sergeyklay** [comment]
I'll take a look ASAP. Thank you for contributting

**Comment by sjinks** [comment]
`if` and `while` accept parentheses for a different reason: the syntax is `if expression` or `while expression`; `expression` can be `( expression )`. That is, parentheses are a part of `expression`.

**Comment by sjinks** [comment]
Implemented in cdf64af

**Comment by phillipdonald** [comment]
Thanks. You guys rock!
